pkg/query/endpointset_test.go

```diff
@@ -336,6 +338,44 @@ func TestEndpointSetUpdate(t *testing.T) {
 			strict:            true,
 			expectedEndpoints: 1,
 		},
+		{
+			name: "long external labels",
+			endpoints: []testEndpointMeta{
+				{
+					InfoResponse: sidecarInfo,
+					extlsetFn: func(addr string) []labelpb.ZLabelSet {
+						sLabel := []string{}
+						for i := 0; i < 1000; i++ {
+							sLabel = append(sLabel, "lbl")
+							sLabel = append(sLabel, "val")
+						}
+						return labelpb.ZLabelSetsFromPromLabels(
+							labels.FromStrings(sLabel...),
+						)
+					},
+				},
+			},
+			expectedEndpoints: 1,
+		},
+		{
+			name: "no external labels",
```

Comment:
This case looks the same as previous one

Reply:
Yes. Only difference is that only store_type is collected collected here (no external labels). I was considering creating a new test func just for optionally specifying labels. This way, it can be extended when we add new labels like record name. What do you think?